proposals/nnnn-screen-manager-alerts.md

```diff
@@ -0,0 +1,231 @@
+# Screen Manager Alert Manager
+
+* Proposal: [SDL-NNNN](NNNN-screen-manager-alerts.md)
+* Author: [Joel Fischer](https://github.com/joeljfischer)
+* Status: **Awaiting review**
+* Impacted Platforms: [iOS / Java Suite / JavaScript Suite]
+
+## Introduction
+This proposal adds alert management (based on the Alert RPC) to the screen manager API.
+
+## Motivation
+The screen manager should be capable of handling all screen related RPCs and features. Template text, template graphics, template soft buttons, menus, and choice interactions are currently handled, but alerts are not.
+
+## Proposed solution
+The proposed solution is to add a new private `SDLAlertManager` sub-manager to the screen manager to handle alert-related capabilities, soft buttons and their images, text concatenation, etc. The `SDLScreenManager` itself will then provide a simple public API for presenting alerts to the screen.
+
+### Audio Processing
+In order to support complicated `Alert`'s complicated audio processing and to simplify `TTSChunk`'s various options, we will add a class that handles initializing all audio-related parameters.
+
+##### iOS
+```objc
+@interface SDLAlertAudioData: NSObject
+
+/**
+ Whether the alert tone should be played before the prompt (if any) is spoken. Defaults to NO. Only works on `Alert`s and will be ignored on other alert types.
```

Comment:
Can you clarify "other alert types"?

Reply:
Removed the line because it's no longer relevant.